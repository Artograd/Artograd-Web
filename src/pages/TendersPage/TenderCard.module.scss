.wrapper {
  border-radius: 3px;
  border: 1px solid red;
  margin: 0 24px;
  box-shadow: 0px 1.9649122953414917px 3.9298245906829834px 0px #1d1e261a;
  box-shadow: 0px 0px 1.9649122953414917px 0px #1d1e260d;
  font-family: var(--uui-font);
}

.content {
  margin: 18px;
  border: 1px solid blue;
}

.optionsCtaWrapper {
  margin-right: 12px;
}

.dimmed {
  color: var(--night600);
}

.meta {
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
}

.title {
  font-size: 18px;
  font-weight: 600;
  line-height: 24px;
  color: var(--night800);
}

.description {
  font-size: 16px;
  font-weight: 400;
  line-height: 24px;
  color: var(--night950);
}
