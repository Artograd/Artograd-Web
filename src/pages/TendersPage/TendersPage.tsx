import {
  Badge,
  Button,
  ControlGroup,
  FlexCell,
  FlexRow,
  FlexSpacer,
  InputAddon,
  Paginator,
  Panel,
  PickerInput,
  SearchInput,
  Text,
} from '@epam/uui';
import styles from './TendersPage.module.scss';
import EmptyFolderIcon from '../../images/emptyFolderIcon.svg';
import { useHistory } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { RootState } from '../../store/store';
import { TenderCard } from '../../components/TenderCard/TenderCard';
import { useEffect, useState } from 'react';
import { CityItemType, Tender, TenderStatus } from '../../types';
import { useArrayDataSource } from '@epam/uui-core';
import { getCityList, getTenders } from '../../requests';

export const TendersPage = () => {
  const history = useHistory();
  const { t } = useTranslation();
  const [isLoading, setIsLoading] = useState(false);
  const [value, onValueChange] = useState<number>(1);
  const [data, setData] = useState<Tender[]>([]);
  const [searchValue, onSearchValueChange] = useState<string>();
  const [listOfCities, setListOfCities] = useState<CityItemType[]>();
  const [statusFilterValue, onStatusFilterValueChange] = useState<
    | {
        id: number;
        name: string;
      }[]
    | unknown[]
  >([]);
  const [locationFilterValue, onLocationFilterValueChange] = useState<
    CityItemType[] | unknown[]
  >([]);
  const userRoles = useSelector(
    (state: RootState) => state?.identity?.userData['cognito:groups'],
  );
  const isOfficer = userRoles?.includes('Officials');
  const statusesList = Object.values(TenderStatus).map((status, index) => ({
    id: index,
    name: status.toLowerCase(),
  }));

  const statusesProvider = useArrayDataSource(
    {
      items: statusesList,
    },
    [],
  );

  const citiesProvider = useArrayDataSource(
    {
      items: listOfCities,
    },
    [],
  );

  const moveAttentionRequiredTender = (tenders: Tender[]) => {
    tenders.map((tender, index) => {
      if (
        tender.status.toLowerCase() === TenderStatus.SELECTION.toLowerCase()
      ) {
        tenders.splice(index, 1);
        tenders.splice(0, 0, tender);
      }
    });
    return tenders;
  };

  useEffect(() => {
    setIsLoading(true);

    getTenders()
      .then((response) => moveAttentionRequiredTender(response.data))
      .then((sortedData) => setData(sortedData));

    getCityList()
      .then((response) => setListOfCities(response))
      .then(() => setIsLoading(false))
      .catch(() => {
        setListOfCities([]);
        setIsLoading(false);
      });
  }, []);

  return (
    <Panel cx={styles.wrapper}>
      <FlexRow>
        <FlexCell width="100%">
          <Text cx={styles.pageTitle}>
            {t('tendersPages.tenders.pageTitle')}
          </Text>
        </FlexCell>
        <FlexSpacer />
        <FlexCell width="100%">
          {data?.length >= 1 && isOfficer && (
            <Button
              color="accent"
              caption={t('tendersPages.tenders.tendersCta')}
              onClick={() => history.push('/tenders/new')}
              rawProps={{ 'data-testid': `header-create-new-tender-cta` }}
              cx={styles.headerTendersCta}
            />
          )}
        </FlexCell>
      </FlexRow>
      <Panel cx={styles.contentWrapper}>
        {data?.length >= 1 && (
          <FlexRow cx={styles.filtersWrapper}>
            <FlexRow cx={styles.creatorFilter}>
              <Badge
                color="info"
                fill="solid"
                caption={t('tendersPages.tenders.allCta')}
              />
              {isOfficer && (
                <Badge
                  color="neutral"
                  fill="solid"
                  caption={t('tendersPages.tenders.createdByMeCta')}
                />
              )}
            </FlexRow>
            <ControlGroup>
              <InputAddon
                content={t('tendersPages.tenders.statusFilterPrefix')}
                cx={styles.prefix}
              />
              <PickerInput
                dataSource={statusesProvider}
                value={statusFilterValue}
                onValueChange={onStatusFilterValueChange}
                getName={(item: { id: number; name: string }) =>
                  t(`global.statuses.${item.name}`)
                }
                entityName="Status filter"
                selectionMode="multi"
                valueType="id"
                inputCx={styles.statusInput}
                placeholder={t('tendersPages.tenders.statusFilterPlaceholder')}
              />
              <InputAddon
                content={t('tendersPages.tenders.cityFilterPrefix')}
                cx={styles.prefix}
              />
              <PickerInput
                dataSource={citiesProvider}
                value={locationFilterValue}
                onValueChange={onLocationFilterValueChange}
                getName={(item: CityItemType) => item.name}
                entityName="Cities filter"
                selectionMode="multi"
                valueType="id"
                inputCx={styles.citiesInput}
                placeholder={t('tendersPages.tenders.cityFilterPlaceholder')}
                sorting={{ field: 'name', direction: 'asc' }}
                isDisabled={isLoading}
              />
            </ControlGroup>
            <FlexSpacer />
            <FlexCell width="auto">
              <SearchInput
                value={searchValue}
                onValueChange={onSearchValueChange}
                placeholder={t('tendersPages.tenders.searchInputPlaceholder')}
                debounceDelay={1000}
                cx={styles.searchField}
              />
            </FlexCell>
          </FlexRow>
        )}
        {data?.length >= 1 &&
          data?.map((tender) => <TenderCard key={tender._id} {...tender} />)}
        {data?.length === 0 && isOfficer && <NoTenders />}
      </Panel>
      {data?.length >= 1 && (
        <FlexRow alignItems="center" cx={styles.paginatorWrapper}>
          <Paginator
            size="24"
            totalPages={1}
            value={value}
            onValueChange={onValueChange}
          />
        </FlexRow>
      )}
    </Panel>
  );
};

const NoTenders = () => {
  const history = useHistory();
  const { t } = useTranslation();
  return (
    <FlexRow>
      <FlexCell cx={styles.tenders} width="100%" textAlign="center">
        <img className={styles.emptyFolderIcon} src={EmptyFolderIcon} />
        <Text cx={styles.tendersTitle}>
          {t('tendersPages.tenders.tendersTitle')}
        </Text>
        <Text cx={styles.tendersDescription}>
          {t('tendersPages.tenders.tendersDescription')}
        </Text>

        <Button
          color="accent"
          caption={t('tendersPages.tenders.tendersCta')}
          onClick={() => history.push('/tenders/new')}
          rawProps={{ 'data-testid': `content-create-new-tender-cta` }}
        />
      </FlexCell>
    </FlexRow>
  );
};
