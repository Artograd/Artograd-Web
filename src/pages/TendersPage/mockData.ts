import { TenderStatus } from '../../types';

export const mockData = [
  {
    _id: '65faecb9ee16c743fc5cfe93',
    title: 'Architectural decoration of the embankment',
    description:
      'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet. Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet.',
    submissionStart: '1711324800000',
    submissionEnd: '1711670400000',
    expectedDelivery: '1711756800000',
    category: ['1', '9', '0'],
    location: {
      nestedLocation: {
        _id: '65f181408de8fcd02bd9930b',
        name: 'Baošići',
      },
      geoPosition: {
        latitude: '42.4429',
        longitude: '18.627',
      },
      addressLine: 'wqertyuiytr 22',
      addressComment: 'wertyuiytr',
    },
    locationLeafId: '65f181408de8fcd02bd9930b',
    showEmail: true,
    files: ['1', '2', '3'],
    snapFiles: [],
    status: TenderStatus.IDEATION,
    ownerName: 'undefined undefined',
    ownerId: 'qwerty',
    createdAt: '1710943414607',
    modifiedAt: '1710943414607',
    organization: 'Testing organization',
    proposals: [
      {
        id: 1,
        published: '1706787480000',
        thumbnail: '',
        selected: false,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Tatiana Press',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
      {
        id: 2,
        published: '1706787480000',
        thumbnail: '',
        selected: false,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Gustavo Vetrovs',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
    ],
  },
  {
    _id: '65faecb9ee16c743fc5cfe93',
    title: 'Architectural decoration of the embankment',
    description:
      'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet. Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet.',
    submissionStart: '1711324800000',
    submissionEnd: '1711670400000',
    expectedDelivery: '1711756800000',
    category: ['1', '9', '0'],
    location: {
      nestedLocation: {
        _id: '65f181408de8fcd02bd9930b',
        name: 'Baošići',
      },
      geoPosition: {
        latitude: '42.4429',
        longitude: '18.627',
      },
      addressLine: 'wqertyuiytr 22',
      addressComment: 'wertyuiytr',
    },
    locationLeafId: '65f181408de8fcd02bd9930b',
    showEmail: true,
    files: ['1', '2', '3'],
    snapFiles: [],
    status: TenderStatus.VOTING,
    ownerName: 'undefined undefined',
    ownerId: 'qwerty',
    createdAt: '1710943414607',
    modifiedAt: '1710943414607',
    organization: 'Testing organization',
    proposals: [
      {
        id: 1,
        published: '1706787480000',
        thumbnail: '',
        selected: false,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Tatiana Press',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
      {
        id: 2,
        published: '1706787480000',
        thumbnail: '',
        selected: false,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Gustavo Vetrovs',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
    ],
  },
  {
    _id: '65faecb9ee16c743fc5cfe93',
    title: 'Architectural decoration of the embankment',
    description:
      'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet. Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet.',
    submissionStart: '1711324800000',
    submissionEnd: '1711670400000',
    expectedDelivery: '1711756800000',
    category: ['1', '9', '0'],
    location: {
      nestedLocation: {
        _id: '65f181408de8fcd02bd9930b',
        name: 'Baošići',
      },
      geoPosition: {
        latitude: '42.4429',
        longitude: '18.627',
      },
      addressLine: 'wqertyuiytr 22',
      addressComment: 'wertyuiytr',
    },
    locationLeafId: '65f181408de8fcd02bd9930b',
    showEmail: true,
    files: ['1', '2', '3'],
    snapFiles: [],
    status: TenderStatus.SELECTION,
    ownerName: 'undefined undefined',
    ownerId: 'qwerty',
    createdAt: '1710943414607',
    modifiedAt: '1710943414607',
    organization: 'Testing organization',
    proposals: [
      {
        id: 1,
        published: '1706787480000',
        thumbnail: '',
        selected: false,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Tatiana Press',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
      {
        id: 2,
        published: '1706787480000',
        thumbnail: '',
        selected: false,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Gustavo Vetrovs',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
      {
        id: 3,
        published: '1706787480000',
        thumbnail: '',
        selected: false,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Corey Ekstrom Bothman',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
    ],
  },
  {
    _id: '65faecb9ee16c743fc5cfe93',
    title: 'Architectural decoration of the embankment',
    description:
      'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet. Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet.',
    submissionStart: '1711324800000',
    submissionEnd: '1711670400000',
    expectedDelivery: '1711756800000',
    category: ['1', '9', '0'],
    location: {
      nestedLocation: {
        _id: '65f181408de8fcd02bd9930b',
        name: 'Baošići',
      },
      geoPosition: {
        latitude: '42.4429',
        longitude: '18.627',
      },
      addressLine: 'wqertyuiytr 22',
      addressComment: 'wertyuiytr',
    },
    locationLeafId: '65f181408de8fcd02bd9930b',
    showEmail: true,
    files: ['1', '2', '3'],
    snapFiles: [],
    status: TenderStatus.CLOSED,
    ownerName: 'undefined undefined',
    ownerId: 'qwerty',
    createdAt: '1710943414607',
    modifiedAt: '1710943414607',
    organization: 'Testing organization',
    proposals: [
      {
        id: 1,
        published: '1706787480000',
        thumbnail: '',
        selected: true,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Tatiana Press',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
      {
        id: 2,
        published: '1706787480000',
        thumbnail: '',
        selected: false,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Gustavo Vetrovs',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
    ],
  },
  {
    _id: '65faecb9ee16c743fc5cfe93',
    title: 'Architectural decoration of the embankment',
    description:
      'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet. Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet.',
    submissionStart: '1711324800000',
    submissionEnd: '1711670400000',
    expectedDelivery: '1711756800000',
    category: ['1', '9', '0'],
    location: {
      nestedLocation: {
        _id: '65f181408de8fcd02bd9930b',
        name: 'Baošići',
      },
      geoPosition: {
        latitude: '42.4429',
        longitude: '18.627',
      },
      addressLine: 'wqertyuiytr 22',
      addressComment: 'wertyuiytr',
    },
    locationLeafId: '65f181408de8fcd02bd9930b',
    showEmail: true,
    files: ['1', '2', '3'],
    snapFiles: [],
    status: TenderStatus.DRAFT,
    ownerName: 'undefined undefined',
    ownerId: 'qwerty',
    createdAt: '1710943414607',
    modifiedAt: '1710943414607',
    organization: 'Testing organization',
    proposals: [],
  },
  {
    _id: '65faecb9ee16c743fc5cfe93',
    title: 'Architectural decoration of the embankment',
    description:
      'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet. Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet.',
    submissionStart: '1711324800000',
    submissionEnd: '1711670400000',
    expectedDelivery: '1711756800000',
    category: ['1', '9', '0'],
    location: {
      nestedLocation: {
        _id: '65f181408de8fcd02bd9930b',
        name: 'Baošići',
      },
      geoPosition: {
        latitude: '42.4429',
        longitude: '18.627',
      },
      addressLine: 'wqertyuiytr 22',
      addressComment: 'wertyuiytr',
    },
    locationLeafId: '65f181408de8fcd02bd9930b',
    showEmail: true,
    files: ['1', '2', '3'],
    snapFiles: [],
    status: TenderStatus.CANCELLED,
    ownerName: 'undefined undefined',
    ownerId: 'qwerty',
    createdAt: '1710943414607',
    modifiedAt: '1710943414607',
    organization: 'Testing organization',
    proposals: [
      {
        id: 1,
        published: '1706787480000',
        thumbnail: '',
        selected: false,
        title: 'Mural depicting an evening stroll',
        description:
          'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit....',
        authors: [
          {
            id: 1,
            authorPic: '',
            authorName: 'Tatiana Press',
            authorRole: 'role',
            authorLinkedIn: 'linkedin',
            authorEmail: 'email',
          },
        ],
      },
    ],
  },
];
