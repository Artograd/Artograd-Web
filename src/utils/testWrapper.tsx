import { renderWithContextAsync } from '@epam/uui-test-utils';
import { MemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { store } from '../store/store';

export const testWrapper = async ({
  component,
  history,
}: {
  component: React.JSX.Element;
  history: MemoryHistory<unknown>;
  
}) => {
  return renderWithContextAsync(<Provider store={store}><Router history={history}>{component}</Router></Provider>);
};
