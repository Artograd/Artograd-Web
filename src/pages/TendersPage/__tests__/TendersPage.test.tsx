import { TendersPage } from '../TendersPage';
import { createMemoryHistory } from 'history';
import { initialState, testWrapper } from '../../../utils/testWrapper';
import { fireEvent, screen } from '@epam/uui-test-utils';
import { RootState } from '../../../store/store';

const initialStateWithOfficerUserData: RootState = {
  ...initialState,
  identity: {
    ...initialState.identity,
    userData: {
      ...initialState.identity.userData,
      'cognito:groups': ['Officials'],
    },
  },
};

describe('Not found page', () => {
  const history = createMemoryHistory();
  test('renders correctly', async () => {
    const component = await testWrapper({
      component: <TendersPage />,
      history,
    });

    expect(component).toMatchSnapshot();
  });

  test('renders correctly for officer role', async () => {
    const component = await testWrapper({
      component: <TendersPage />,
      history,
      state: initialStateWithOfficerUserData,
    });

    expect(component).toMatchSnapshot();
  });

  test('redirect on create new tender page on header button click', async () => {
    const history = createMemoryHistory();

    await testWrapper({
      component: <TendersPage />,
      history,
      state: initialStateWithOfficerUserData,
    });

    fireEvent.click(screen.getByTestId('header-create-new-tender-cta'));

    expect(history.location.pathname).toBe('/tenders/new');
  });

  test('redirect on create new tender page on button click when no tenders', async () => {
    const history = createMemoryHistory();

    await testWrapper({
      component: <TendersPage />,
      history,
      state: initialStateWithOfficerUserData,
    });

    fireEvent.click(screen.getByTestId('content-create-new-tender-cta'));

    expect(history.location.pathname).toBe('/tenders/new');
  });
});
