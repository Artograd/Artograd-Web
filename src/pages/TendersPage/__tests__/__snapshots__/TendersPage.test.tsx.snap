// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not found page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="uui-panel nNCe-- wrapper uEZmHy"
      >
        <div
          class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
        >
          <div
            class=""
            style="min-width: 0; flex-basis: 100%;"
          >
            <div
              class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV pageTitle doyi1y"
            >
              Tenders
            </div>
          </div>
          <div
            class="CcZAjD"
          />
          <div
            class=""
            style="min-width: 0; flex-basis: 100%;"
          />
        </div>
        <div
          class="uui-panel nNCe-- contentWrapper uEZmHy"
        >
          <div
            class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="uui-panel nNCe-- wrapper uEZmHy"
    >
      <div
        class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
      >
        <div
          class=""
          style="min-width: 0; flex-basis: 100%;"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV pageTitle doyi1y"
          >
            Tenders
          </div>
        </div>
        <div
          class="CcZAjD"
        />
        <div
          class=""
          style="min-width: 0; flex-basis: 100%;"
        />
      </div>
      <div
        class="uui-panel nNCe-- contentWrapper uEZmHy"
      >
        <div
          class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByAria": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByRoleAndText": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByAria": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByRoleAndText": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByAria": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByRoleAndText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByAria": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByRoleAndText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByAria": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByRoleAndText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByAria": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByRoleAndText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Not found page renders correctly for officer role 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="uui-panel nNCe-- wrapper uEZmHy"
      >
        <div
          class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
        >
          <div
            class=""
            style="min-width: 0; flex-basis: 100%;"
          >
            <div
              class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV pageTitle doyi1y"
            >
              Tenders
            </div>
          </div>
          <div
            class="CcZAjD"
          />
          <div
            class=""
            style="min-width: 0; flex-basis: 100%;"
          >
            <button
              class="uui-button-box uui-enabled -clickable VFUd6V y6sOzI uui-button uui-fill-solid uui-color-accent uui-size-36 headerTendersCta"
              data-testid="header-create-new-tender-cta"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Create New Tender
              </div>
            </button>
          </div>
        </div>
        <div
          class="uui-panel nNCe-- contentWrapper uEZmHy"
        >
          <div
            class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
          >
            <div
              class="tenders"
              style="min-width: 0; flex-basis: 100%; text-align: center;"
            >
              <img
                class="emptyFolderIcon"
                src="emptyFolderIcon.svg"
              />
              <div
                class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV tendersTitle doyi1y"
              >
                Create your first tender.
              </div>
              <div
                class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV tendersDescription doyi1y"
              >
                Creating a tender, you announce open calls for street art, invite sculptors, artists, and creators to enhance the cityscape.
              </div>
              <button
                class="uui-button-box uui-enabled -clickable VFUd6V y6sOzI uui-button uui-fill-solid uui-color-accent uui-size-36"
                data-testid="content-create-new-tender-cta"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Create New Tender
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="uui-panel nNCe-- wrapper uEZmHy"
    >
      <div
        class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
      >
        <div
          class=""
          style="min-width: 0; flex-basis: 100%;"
        >
          <div
            class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV pageTitle doyi1y"
          >
            Tenders
          </div>
        </div>
        <div
          class="CcZAjD"
        />
        <div
          class=""
          style="min-width: 0; flex-basis: 100%;"
        >
          <button
            class="uui-button-box uui-enabled -clickable VFUd6V y6sOzI uui-button uui-fill-solid uui-color-accent uui-size-36 headerTendersCta"
            data-testid="header-create-new-tender-cta"
            tabindex="0"
            type="button"
          >
            <div
              class="uui-caption"
            >
              Create New Tender
            </div>
          </button>
        </div>
      </div>
      <div
        class="uui-panel nNCe-- contentWrapper uEZmHy"
      >
        <div
          class="uui-flex-row kfXZUl _7f-LqE _5Jq2IB RkwqFu"
        >
          <div
            class="tenders"
            style="min-width: 0; flex-basis: 100%; text-align: center;"
          >
            <img
              class="emptyFolderIcon"
              src="emptyFolderIcon.svg"
            />
            <div
              class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV tendersTitle doyi1y"
            >
              Create your first tender.
            </div>
            <div
              class="Ka60RD uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography CMzhku xhqm1P jzu7jV tendersDescription doyi1y"
            >
              Creating a tender, you announce open calls for street art, invite sculptors, artists, and creators to enhance the cityscape.
            </div>
            <button
              class="uui-button-box uui-enabled -clickable VFUd6V y6sOzI uui-button uui-fill-solid uui-color-accent uui-size-36"
              data-testid="content-create-new-tender-cta"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Create New Tender
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByAria": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByRoleAndText": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByAria": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByRoleAndText": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByAria": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByRoleAndText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByAria": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByRoleAndText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByAria": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByRoleAndText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByAria": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByRoleAndText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
